src/util/system.cpp

```diff
@@ -1063,13 +1065,9 @@ void ArgsManager::LogArgs() const
 
 bool RenameOver(fs::path src, fs::path dest)
 {
-#ifdef WIN32
-    return MoveFileExW(src.wstring().c_str(), dest.wstring().c_str(),
-                       MOVEFILE_REPLACE_EXISTING) != 0;
-#else
-    int rc = std::rename(src.string().c_str(), dest.string().c_str());
-    return (rc == 0);
-#endif /* WIN32 */
+    std::error_code error;
+    std::filesystem::rename(src.string(), dest.string(), error);
```

Comment:
In commit "util: use stronger-guarantee rename method" (99b151df9c074f60150c5eb777ed98bfd7118e91)

This will result in narrow string conversion on windows, and after #22937 will be a compiler error. Should be easy to fix by changing to:

```c++
std::filesystem::rename(src.native(), dest.native(), error);
```

This should also work after #20744, but after #20744 it could be simplified more to

```c++
std::filesystem::rename(src, dest, error);
```

Reply:
Used just `src` and `dest`, thanks!